src/VR.js

```diff
@@ -126,9 +126,12 @@ class GamepadHapticActuator {
     });
   }
 }
+    var index = 0;
 class Gamepad {
   constructor(id, hand, xrGamepad, hapticActuator) {
     this.id = id;
+    this.index = index;
+    index++;
```

Comment:
I would probably put this as part of the constuctor parameters. I expect engines to be sensitive to the actual value in cases where e.g. the gamepads are reconstructed, and it could be expected that the indices always start at zero. even when that happens.

Reply:
Resolved in https://github.com/exokitxr/exokit/pull/1390/commits/b3054dbd89b024fba548af02380f487c1890d409